### PR TITLE
[Backport 2.16] Remove hardcoded line number from the zeroize test

### PR DESCRIPTION
**Pull request by bensze01**
This is the 2.16 backport of #3363. The only changes are in the first commit, which was needed for the patch to apply cleanly. It reverts some superfluous changes introduced in #2710, as indicated in the commit message

### CODE REVIEW

File: programs/test/zeroize.c [hunk 1]
```diff
@@ -40,7 +38,6 @@
 #if defined(MBEDTLS_PLATFORM_C)
 #include "mbedtls/platform.h"
 #else
-#include <stdio.h>
```
Comment (danh-arm) [inline programs/test/zeroize.c:43]: Why this change in the backport but not the main PR? It looks off-topic (though I guess harmless).

Comment (bensze01) [reply programs/test/zeroize.c:43]: As mentioned in the commit message for the first commit, this line was never present in the development branch. It was erroneously added to the file, causing the two branches to diverge.
This also changed the line number hardcoded in the GDB script, causing my patch to not apply cleanly to the 2.16 branch.

Comment (danh-arm) [reply programs/test/zeroize.c:43]: Thanks - I failed to look at the individual commits before, just the overall change.
